app/models/reservation.rb

```diff
@@ -73,8 +73,12 @@ def confirmed?
     status == Status::CONFIRMED
   end
 
+  def cancelable?
+    (Status::CANCELABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
+  end
+
   def modifiable?
-    Status::MODIFIABLE.include? status
+    (Status::MODIFIABLE.include? status) && ((self.start.to_i-Time.now.to_i) > (self.nook.modifiable_before*3600))
```

Comment:
@shubhpatel108 To fix that a simple condition to restrict the button in the views should suffice, I'll push another commit fixing this in #143 .